csrc/flash_attn/flash_api.cpp

```diff
@@ -524,7 +526,7 @@ mha_varlen_fwd(const at::Tensor &q,  // total_q x num_heads x head_size, total_q
 
     auto opts = q.options();
 
-    auto softmax_lse = torch::empty({batch_size, num_heads, max_seqlen_q}, opts.dtype(at::kFloat));
+    auto softmax_lse = torch::empty({num_heads, total_q}, opts.dtype(at::kFloat));
```

Comment:
The arguments for device be passed in to allocate on device?
```suggestion
    auto softmax_lse = torch::empty({num_heads, total_q}, dtype=opts.dtype(at::kFloat), device=q.get_device());
```